### PR TITLE
Implement SDL 0099, 0165, 0182 RC LIGHT, AUDIO, HMI_SETTINGS and parameters (SIS Data),  More Names and Status Values,  Audio Source AM/FM/XM/DAB

### LINKED ISSUE #755: [SDL 0099] New remote control modules (LIGHT, AUDIO, HMI_SETTINGS) and parameters (SIS Data)
Proposal: [New remote control modules (LIGHT, AUDIO, HMI_SETTINGS) and parameters (SIS Data)](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0099-new-remote-control-modules-and-parameters.md)

> SDL remote control provides the ability for mobile applications to control certain components of the vehicle, such as, tune the radio, set the desired temperature or change the fan speed. However, currently only `CLIMATE` and `RADIO` modules are available for remote-control. Mobile application developers need more than basic climate or radio control. To meet developers' requirements, we propose to add three new modules (`AUDIO`, `LIGHT`, `HMI_SETTINGS`) and some new parameters (heatedXXXEnable and radio sisData) to the existing `CLIMATE` and `RADIO` modules respectively. This proposal only defines new data types and new parameters added to the existing data types. It does not add or change any remote control RPCs.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/296

Steering Committee Decision:

> The Steering Committee voted to accept this proposal with revisions. The author will revise the proposal to fix a few typos, as well as include descriptions for the groups of lights - listing which single light elements they typically include. It was also noted that limiting the scope of proposals, especially those related to Remote Control, in the future will help to make for easier understanding and discussion among SDLC Members.
>Proposal has now been revised to include the descriptions and fix the typos mentioned.

### LINKED ISSUE #1022: [SDL 0182] Audio Source AM/FM/XM/DAB
Proposal: [Audio Source AM/FM/XM/DAB](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0182-audio-source-am-fm-xm.md)

> SDL remote control allows a mobile application to change the current audio source. When an application sets the audio source to `RADIO_TUNER`, the vehicle is supposed to use the last known/used radio band (AM/FM/XM) and frequency/station of the tuner. However, the application has no knowledge of last used radio band before sending such a request. The vehicle may or may not store the last used radio band. Therefore, the result of setting the audio source to `RADIO_TUNER` is unknown to the application. It is better for an application to set the audio source directly to `AM`, `FM`, `XM` (`XM` is for Sirius XM) or `DAB` (digital audio broadcasting, including DAB+).

Review: https://github.com/smartdevicelink/sdl_evolution/issues/532

Steering Committee Decision:

> The Steering Committee voted to accept this proposal, with the following revision: the Motivation section of the proposal should reference that this is an update to proposal SDL 0099, and include a link to that proposal.
> The proposal was revised to incorporate these revisions on July 11, 2018.

### PR DESCRIPTION
**Pull request by mrapitis**
Fixes #755, #954, #1022 

This PR is **ready** for review.

## Risk

This PR makes **minor** API changes.

## Testing Plan

Added new unit tests and expanded existing ones

## Summary
Added RPC's, structs, enums as defined in Remote Control - LIGHT, AUDIO, HMI_SETTINGS and parameters (SIS Data)  proposal, Remote Control – Lights – More Names and Status Values, Audio Source AM/FM/XM/DAB proposal, and 

## CLA

- [x] I have signed the CLA

### CODE REVIEW

**Comment by mrapitis** [comment]
@joeljfischer PR has been updated to include #755, #954, and #1022.  Please review when time permits. Thanks!

File: SmartDeviceLink/SDLLightState.h [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+@interface SDLLightState : SDLRPCStruct
+
+/**
+ Constructs a newly allocated SDLLightState object with given parameters
+
+ @param id The name of a light or a group of lights
+ @param status Reflects the status of Light.
+ @return An instance of the SDLLightState class
+ */
+- (instancetype)initWithID:(SDLLightName)id status:(SDLLightStatus)status;
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLLightState.h]: Change to `initWithId...`

File: SmartDeviceLink/SDLLightState.h [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+/**
+ Constructs a newly allocated SDLLightState object with given parameters
+
+ @param id The name of a light or a group of lights
+ @param status Reflects the status of Light.
+ @param density Reflects the density of Light.
+ @param sRGBColor Reflects the color of Light.
+ @return An instance of the SDLLightState class
+ */
+- (instancetype)initWithID:(SDLLightName)id status:(SDLLightStatus)status density:(double)density sRGBColor:(SDLSRGBColor *)sRGBColor;
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLLightState.h]: Change to `initWithId...`

File: SmartDeviceLink/SDLRadioControlCapabilities.h [hunk 3]
```diff
@@ -26,7 +26,25 @@ NS_ASSUME_NONNULL_BEGIN
...
+- (instancetype)initWithModuleName:(NSString *)moduleName radioEnableAvailable:(BOOL)radioEnableAvailable radioBandAvailable:(BOOL)radioBandAvailable radioFrequencyAvailable:(BOOL)radioFrequencyAvailable hdChannelAvailable:(BOOL)hdChannelAvailable rdsDataAvailable:(BOOL)rdsDataAvailable availableHDsAvailable:(BOOL)availableHDsAvailable stateAvailable:(BOOL)stateAvailable signalStrengthAvailable:(BOOL)signalStrengthAvailable signalChangeThresholdAvailable:(BOOL)signalChangeThresholdAvailable __deprecated_msg(("Use initWithModuleName:moduleName:radioEnableAvailable radioFrequencyAvailable:hdChannelAvailable:rdsDataAvailable:availableHDsAvailable:stateAvailable:signalStrengthAvailable:signalChangeThresholdAvailable:hdRadioEnableAvailable:siriusXMRadioAvailable:sisDataAvailable: instead"));
+/**
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLRadioControlCapabilities.h]: Add a new line between these

File: SmartDeviceLink/SDLRadioControlCapabilities.m [hunk 4]
```diff
@@ -17,6 +17,11 @@ - (instancetype)initWithModuleName:(NSString *)moduleName radioEnableAvailable:(
 }
 
 - (instancetype)initWithModuleName:(NSString *)moduleName radioEnableAvailable:(BOOL)radioEnableAvailable radioBandAvailable:(BOOL)radioBandAvailable radioFrequencyAvailable:(BOOL)radioFrequencyAvailable hdChannelAvailable:(BOOL)hdChannelAvailable rdsDataAvailable:(BOOL)rdsDataAvailable availableHDsAvailable:(BOOL)availableHDsAvailable stateAvailable:(BOOL)stateAvailable signalStrengthAvailable:(BOOL)signalStrengthAvailable signalChangeThresholdAvailable:(BOOL)signalChangeThresholdAvailable hdRadioEnableAvailable:(BOOL)hdRadioEnableAvailable siriusXMRadioAvailable:(BOOL)siriusXMRadioAvailable {
+
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLRadioControlCapabilities.m]: Remove the extra line

File: SmartDeviceLink/SDLRadioControlCapabilities.h [hunk 5]
```diff
@@ -43,9 +61,10 @@ NS_ASSUME_NONNULL_BEGIN
...
+- (instancetype)initWithModuleName:(NSString *)moduleName radioEnableAvailable:(BOOL)radioEnableAvailable radioBandAvailable:(BOOL)radioBandAvailable radioFrequencyAvailable:(BOOL)radioFrequencyAvailable hdChannelAvailable:(BOOL)hdChannelAvailable rdsDataAvailable:(BOOL)rdsDataAvailable availableHDsAvailable:(BOOL)availableHDsAvailable stateAvailable:(BOOL)stateAvailable signalStrengthAvailable:(BOOL)signalStrengthAvailable signalChangeThresholdAvailable:(BOOL)signalChangeThresholdAvailable hdRadioEnableAvailable:(BOOL)hdRadioEnableAvailable siriusXMRadioAvailable:(BOOL)siriusXMRadioAvailable sisDataAvailable:(BOOL)sisDataAvailable;
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLRadioControlCapabilities.h]: If this "new" method is just modifying a method that has never actually been released (which it seems to be, I can't find this deprecated method on the master branch), you can just modify the method directly.

File: SmartDeviceLink/SDLRemoteControlCapabilities.h [hunk 6]
```diff
@@ -16,18 +19,32 @@ NS_ASSUME_NONNULL_BEGIN
...
- Constructs a newly allocated SDLRemoteControlCapabilities object with cushion and firmness
+ Constructs a newly allocated SDLRemoteControlCapabilities object with given parameters
+
+ @param climateControlCapabilities Array of SDLClimateControlCapabilities
+ @param radioControlCapabilities Array of SDLRadioControlCapabilities
+ @param buttonCapabilities Array of SDLButtonCapabilities
+ @param seatControlCapabilities Array of SDLSeatControlCapabilities
+ @return An instance of the SDLRemoteControlCapabilities class
+ */
+- (instancetype)initWithClimateControlCapabilities:(nullable NSArray<SDLClimateControlCapabilities *> *)climateControlCapabilities radioControlCapabilities:(nullable NSArray<SDLRadioControlCapabilities *> *)radioControlCapabilities buttonCapabilities:(nullable NSArray<SDLButtonCapabilities *> *)buttonCapabilities seatControlCapabilities:(nullable NSArray<SDLSeatControlCapabilities *> *)seatControlCapabilities __deprecated_msg("Use initWithClimateControlCapabilities:climateControlCapabilities:radioControlCapabilities:buttonCapabilities:seatControlCapabilities:audioControlCapabilities:hmiSettingsControlCapabilities:lightControlCapabilities: instead");
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLRemoteControlCapabilities.h]: Same as above, this seems to not exist on master

**Comment by mrapitis** [comment]
@joeljfischer please re-review when time permits. Thanks!

File: SmartDeviceLink/SDLLightName.h [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+ * The name that identifies the Light
+ *
+ */
+typedef SDLEnum  SDLLightName SDL_SWIFT_ENUM;
+
+/**
+ * @abstract Represents the Light with name FRONT_LEFT_HIGH_BEAM.
+ *
+ */
+extern  SDLLightName const  SDLLightNameFrontLeftHighBeam;
```
[the remark this answers is not in the document]
Comment (joeljfischer) [reply SmartDeviceLink/SDLLightName.h]: @mrapitis The spacing is still odd here. Two spaces between words in many locations

**Comment by joeljfischer** [comment]
@mrapitis The spacing in SDLLightName.h is still wonky. Many words have two spaces between them

**Pull request merged**